### PR TITLE
Azure Service Principal for Azure CLI Rewrite

### PR DESCRIPTION
**Pull request by daphnemamsft**

### CODE REVIEW

**Comment by dbradish-microsoft** [comment]
@daphnemamsft , This PR looks like a WIP as I believe you are working to break this Tutorial down into multiple files, refresh and add possible content to each section (like output code blocks)?  I won't review until you are ready.

File: docs-ref-conceptual/azure-cli-sp-tutorial-4.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ms.author: daphnema
+ms.date: 09/6/2023
+ms.topic: conceptual
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli, seo-azure-cli
+keywords: azure service principal, create service principal azure, create service principal azure cli
+---
+
+# 4 - Using Service Principal with password-based authentication
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-sp-tutorial-4.md:15]: Let's change this to something like `Create a service principal using password-based authentication`.  Also, do a quick check that "service principal" isn't capitalized unless "Service" comes at the beginning of a sentence.

File: docs-ref-conceptual/azure-cli-sp-tutorial-5.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+ms.author: daphnema
+ms.date: 08/16/2023
+ms.topic: conceptual
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli, seo-azure-cli
+keywords: azure service principal, create service principal azure, create service principal azure cli
+---
+
+# 5 - Using Service Principal with certificate-based authentication
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-sp-tutorial-5.md:15]: Let's change this to something like `Create a service principal using certificate-based authentication`

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]
A customer has to create a SP before they can "get" or "manage roles", so let's move 4 & 5 to be 2 & 3, and 2 & 3 down into the 4&5 slots.

Also please review https://review.learn.microsoft.com/en-us/help/patterns/level4/article-tutorial?branch=main and make sure we have followed Contributor guidelines.  Thank you!

File: docs-ref-conceptual/azure-cli-sp-tutorial-1.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+> * Use service principals with a password.
+> * Use service principals with a certificate.
+> * Get an existing service principal.
+> * Manage service principal roles
+> * Create a resource using service principal
+
+## Prerequisites
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/azure-cli-sp-tutorial-1.md:33]: Prerequisites need to follow the azCLI [guidelines](https://review.learn.microsoft.com/en-us/help/contribute/conventions-azure-cli?branch=main#prerequisites).  Here is another example from inside this same repo: https://review.learn.microsoft.com/en-us/cli/azure/azure-cli-vm-tutorial-1?branch=pr-en-us-3798#prerequisites. It is also important to talk about the permissions needed in this prerequisite section.

Your markdown would look like this:

## Prerequisites
* To create a service principal, you must have ... <what permissions?>
[!INCLUDE [include](~/articles/reusable-content/azure-cli/azure-cli-prepare-your-environment-no-header.md)]

File: docs-ref-conceptual/azure-cli-sp-tutorial-3.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+To learn more about why and how certificate based authentication is more secure, see [Azure Active Directory certificate based authentication concepts](/azure/active-directory/authentication/concept-certificate-based-authentication). 
+
+## How do I add certificates to a Service Principal?
+
+We recommend you use Azure CLI to perform certificate-based authentication, however there is also a way to manually assign an existing certificate to a service principal through the Azure portal:
+
+# [Azure CLI](#tab/concepts)
+
+For certificate-based authentication, use the `--cert` parameter. This parameter requires that you hold an existing certificate. Make sure any tool that uses this service principal has access to the certificate's private key. Certificates should be in an ASCII format such as PEM, CER, or DER. Pass the certi897ficate as a string, or use the `@path` format to load the certificate from a file.
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-sp-tutorial-3.md:31]: Why is `certificate` spelled as `certi897ficate`?

File: docs-ref-conceptual/azure-cli-sp-tutorial-3.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+> [!NOTE]
+> When using a PEM file, the **CERTIFICATE** must be appended to the **PRIVATE KEY** within the file.
```
Comment (jiasli) [inline docs-ref-conceptual/azure-cli-sp-tutorial-3.md:34]: This statement is wrong. The private key should not be passed: https://github.com/MicrosoftDocs/azure-docs-cli/issues/4734

**Pull request merged**